src/extensions/replay/sessionrecording.ts

```diff
@@ -863,7 +863,11 @@ export class SessionRecording {
         }
 
         if (this.buffer.data.length > 0) {
-            const snapshotEvents = splitBuffer(this.buffer)
+            const captureEventsPerSecond = this.instance.rateLimiter.captureEventsPerSecond
+            const snapshotEvents = splitBuffer(this.buffer, captureEventsPerSecond)
+            if (snapshotEvents.length >= captureEventsPerSecond) {
```

Comment:
🤯 oh... knowing that recording is sending more events than, erm , events. We should either not limit them client side or have a much higher limit 🤔